### PR TITLE
Handle customer tests

### PR DESCRIPTION
**Pull request by rehanjaved237-zz**

### CODE REVIEW

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-dac7da5a`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-c74192b9`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-7d61cbd4`

File: controllers/project_controller_test.go [hunk 1]
```diff
@@ -110,6 +114,10 @@ var _ = Describe("ProjectController", func() {
 
 		projectInvalidInput := mockData.CreateProjectInvalidInput
 
+		AfterEach(func() {
+			util.TryDeleteProject(projectInvalidInput.Spec.Name, ns)
```
Comment (ahmedwaleedmalik) [inline controllers/project_controller_test.go]: Why are we trying to delete `projectInvalidInput.Spec.Name` in afterEach block ?

File: controllers/customer_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stakater/jira-service-desk-operator/api/v1alpha1"
+	mockData "github.com/stakater/jira-service-desk-operator/mock"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("Customer Controller", func() {
+
```
Comment (ahmedwaleedmalik) [inline controllers/customer_controller_test.go:17]: Ideally this should have a `beforeAll` block to create a project and `afterAll` block to delete that project

File: controllers/project_controller.go [hunk 3]
```diff
@@ -116,6 +116,7 @@ func (r *ProjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			}
 		}
 	}
+
```
Comment (ahmedwaleedmalik) [inline controllers/project_controller.go:119]: Remove this redundant newline

File: pkg/jiraservicedesk/client/client_customer.go [hunk 4]
```diff
@@ -72,6 +73,8 @@ func (c *jiraServiceDeskClient) GetCustomerById(customerAccountId string) (Custo
 }
 
 func (c *jiraServiceDeskClient) CreateCustomer(customer Customer) (string, error) {
+	fmt.Println("Check1")
```
Comment (ahmedwaleedmalik) [inline pkg/jiraservicedesk/client/client_customer.go]: Remove this log

File: pkg/jiraservicedesk/client/client_customer.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package client
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
```
Comment (ahmedwaleedmalik) [inline pkg/jiraservicedesk/client/client_customer.go]: Remove this import

File: pkg/jiraservicedesk/client/client_project_test.go [hunk 6]
```diff
@@ -43,7 +42,7 @@ func TestJiraService_GetProject_shouldNotGetProject_whenInValidProjectIdIsGiven(
 	jiraClient := NewClient("", mockData.BaseURL, "")
 	_, err := jiraClient.GetProjectById("/" + mockData.ProjectID)
 
-	st.Expect(t, err, errors.New(mockData.GetProjectFailedErrorMsg))
+	st.Reject(t, err, nil)
```
Comment (ahmedwaleedmalik) [inline pkg/jiraservicedesk/client/client_project_test.go]: Please use `Expect` instead of `Reject` as discussed

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-b72cdc82`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-f0b9af78`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-c26c4d6f`

**Review by ahmedwaleedmalik — APPROVED** [review]
LGTM

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/jira-service-desk-operator:SNAPSHOT-PR-36-136fd1bc`

**Pull request merged**